### PR TITLE
Split integration tests into different steps 

### PR DESCRIPTION
**Pull request by chkeita**
The PR refactors the integration test to help us build a pipeline with multiple stages that can be retried

- Refactoring check-pr.py to extract the logic of downloading the binaries
- refactoring integration-tets.py to split the logic of `setup`, `launch`, `check_result` and `cleanup`

### CODE REVIEW

File: src/utils/check-pr/github_client.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+            time.sleep(frequency)
+            frames.sort(key=frames[0].__eq__)
+    finally:
+        if waited and isatty:
+            print(flush=True)
+
+    return result[2]
+
+
+class GithubClient:
```
Comment (chkeita) [inline src/utils/check-pr/github_client.py:56]: This is just the Downloaded class that existed in check-pr.py
I just renamed it to `GithubClient` without changing the logic

**Review by stishkin — APPROVED** [review]

File: src/utils/check-pr/github_client.py (hunk 1 above)
Comment (nharper285) [reply src/utils/check-pr/github_client.py:56]: Is there any new code here? Or is it all pulled from the original check-pr

Comment (chkeita) [reply src/utils/check-pr/github_client.py:56]: no new code.

**Review by nharper285 — APPROVED** [review]
Looks good! Can't wait to try out.

**Pull request merged**